include/ethash/ethash.hpp

```diff
@@ -128,13 +140,13 @@ inline result hash(
 
 result hash(const epoch_context_full& context, const hash256& header_hash, uint64_t nonce) noexcept;
 
-inline bool verify_final_hash(const hash256& header_hash, const hash256& mix_hash, uint64_t nonce,
-    const hash256& boundary) noexcept
+inline std::error_code verify_final_hash(const hash256& header_hash, const hash256& mix_hash,
```

Comment:
Did you mean to use `ethash_errc` here ?
std::error_code is generally used for OS errors.
I believe overall returning an `int` from enum is enough instead of using the complexity of std::error_category

Just thinking out loud

Reply:
No, I wanted this to standardize error handling in C++. There is a separate commit which adds support for it. This additionally adds you ability to print the error message (see https://github.com/torquem-ch/silkworm/pull/444/files#diff-b2566784e825a40094310b333efb70a061749da92037515ebfec3398052f0461R143).